### PR TITLE
CMIniFile: Added DistributionPointInstallIis param

### LINKED ISSUE #62: CMIniFile: Missing parameter for DistributionPointInstallIIS
#### Details of the scenario you tried and the problem that is occurring

When including the settings to install the distribution point role in the CMIniFile configuration, the DistributionPointInstallIIS setting is not currently implemented in the resource and appears to be required by the SCCM/MECM installer. 

#### Verbose logs showing the problem
When running the SCCM CLI install without the DistributionPointInstallIIS option included in the .ini file, the messages below occur in the ConfigMgrSetupWizard log:

9/2/2020 2:04:57 PM: ConfigMgrSetupWizard Verbose: 1 : Scripted setup failure handling callback called. Failure: 103.
9/2/2020 2:05:08 PM: ConfigMgrSetupWizard Verbose: 1 : Received error [Could not connect or execute SQL query.] from native code.
9/2/2020 2:05:08 PM: ConfigMgrSetupWizard Verbose: 1 : Exiting SetupWPF.exe.
#### Suggested solution to the issue
Add boolean parameter for DistributionPointInstallIIS to CMIniFile resource if distribution point role is specified in the configuration
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
# insert configuration here
        # Provision .ini file
        CMIniFile AGMLabIni
        {
            IniFileName               = 'AGT-Lab-Pri.ini'
            IniFilePath               = $env:TEMP
            Action                    = 'InstallPrimarySite'
            ProductID                 = 'EVAL'
            SiteCode                  = 'AGT'
            SiteName                  = 'SCCM-Primary Site'
            SMSInstallDir             = "${disk}:\Program Files\Microsoft Configuration Manager"
            SDKServer                 = "$env:COMPUTERNAME.$DomainName"
            PreRequisiteComp          = $true
            PreRequisitePath          = "$path\SCCM\SCCMPrereqs"
            AdminConsole              = $true
            JoinCeip                  = $false
            MobileDeviceLanguage      = $false
            RoleCommunicationProtocol = 'HTTPorHTTPS'
            ClientsUsePKICertificate  = $false
            ManagementPoint           = "$env:COMPUTERNAME.$DomainName"
            ManagementPointProtocol   = 'HTTP'
            DistributionPoint         = "$env:COMPUTERNAME.$DomainName"
            DistributionPointProtocol = 'HTTP'
            SQLServerName             = "$sqlServer.$DomainName"
            DatabaseName              = "DB_Name"
            SQLSSBPort                = "4022"
            SQLLogFilePath            = 'S:\SQLLogs\'
            SQLDataFilePath           = 'S:\SQLDB\'
            CloudConnector            = $false
            CurrentBranch             = $true
            SAActive                  = $true
        }


#### The operating system the target node is running
OsName               : Microsoft Windows Server 2016 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.3866.amd64fre.rs1_release.200805-1327
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version and build of PowerShell the target node is running
Name                           Value                                                                                              
----                           -----                                                                                              
PSVersion                      5.1.14393.3866                                                                                     
PSEdition                      Desktop                                                                                            
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                            
BuildVersion                   10.0.14393.3866                                                                                    
CLRVersion                     4.0.30319.42000                                                                                    
WSManStackVersion              3.0                                                                                                
PSRemotingProtocolVersion      2.3                                                                                                
SerializationVersion           1.1.0.1  

#### Version of the DSC module that was used
0.2.0

### PR DESCRIPTION
**Pull request by rchristman89**
#### Pull Request (PR) description
Added DistributionPointInstallIis paramater to the resource.
Throw an error in set if DistributionPoint is provided with out this paramater and write a warning in Test.

#### This Pull Request (PR) fixes the following issues
- Fixes #62 

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/70)

### CODE REVIEW

File: source/DSCResources/DSC_CMIniFile/DSC_CMIniFile.psm1 [hunk 1]
```diff
@@ -682,34 +696,39 @@ function Set-TargetResource
     {
         throw ($script:localizedData.ProxyError)
     }
+    elseif ($DistributionPoint -and (-not $DistributionPointInstallIis))
+    {
+        throw $script:localizedData.DistributionPointError
```
Comment (NEllis280) [inline source/DSCResources/DSC_CMIniFile/DSC_CMIniFile.psm1:701]: Use parenthesis here for consistency

**Comment by NEllis280** [comment]
I have tested this PR and everything works as I expect. Only one outstanding comment from code review.

File: source/DSCResources/DSC_CMIniFile/DSC_CMIniFile.psm1 (hunk 1 above)
Comment (rchristman89) [reply source/DSCResources/DSC_CMIniFile/DSC_CMIniFile.psm1:701]: Fixed.

**Review by NEllis280 — APPROVED** [review]
Code looks good and tests as expected

**Pull request merged**